pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -695,6 +695,23 @@ func getServersConfig(cr *api.KieApp) ([]api.ServerTemplate, error) {
 			}
 
 			// Set replicas
+			if len(cr.Spec.Objects.Servers) == 0 {
+				serverSet.Replicas = serverReplicas
+			} else {
+				for i := range cr.Spec.Objects.Servers {
+					server := cr.Spec.Objects.Servers[i]
+					// if the configuration use only one replica we increment
+					if server.JBPMCluster && *serverReplicas == 1 {
+						serverReplicasWithJBPMCluster := *serverReplicas
+						if serverSet.Replicas == nil {
+							serverReplicasWithJBPMCluster = *serverReplicas + 1
+						} else {
+							serverReplicasWithJBPMCluster = *serverSet.Replicas + 1
+						}
+						serverSet.Replicas = &serverReplicasWithJBPMCluster
+					}
+				}
```

Comment:
At this point of the code base you're already in the servers range. instead of doing this again, i wonder if the existing serverSet object would do the job.

Reply:
Removed iteration

Reply:
lines 706-709 should probably be moved into the existing replicas check below.

Reply:
once you do that, then the following line is unnecessary and can be removed.
`serverSet.Replicas = new(int32)`